### PR TITLE
Add auto-tune for the softmax kernel in benchmark

### PR DESCRIPTION
**Pull request by chengjunlu**
1. Fix the bus in the auto-tune of the softmax kernel in benchmark.
2. Pick up the change from @victor-eds adding the threads_per_warp in autotuner.

Note: The Triton softmax kernel in the tutorial may not get best performance on Intel GPU for small N sizes.
Because it uses the one work group to compute softmax for one single row. It is not efficient on Intel GPU for small N (<1024).
Unlike to NV GPU, Intel GPU has different threads dispatching and paralleling in HW. 
We need the kernel to explicitly compute the softmax on a matrix instead of a single row. (One work group compute multiple rows when the N is smaller than 1024)

### CODE REVIEW

**Review by ESI-SYD — APPROVED** [review]

**Comment by whitneywhtsang** [comment]
threads_per_warp is added in our intel fork, and doesn't exist in upstream: https://github.com/intel/intel-xpu-backend-for-triton/pull/414

PyTorch/IPEX can specify warp_size when passing in the target: https://github.com/intel/intel-extension-for-pytorch/blob/dev/triton-test-3.0/intel_extension_for_pytorch/_inductor/xpu/triton_heuristics.py#L133. The warp_size GPUTarget field comes from upstream, can we autotune that instead?

**Comment by chengjunlu** [comment]
> threads_per_warp is added in our intel fork, and doesn't exist in upstream: #414
> 
> PyTorch/IPEX can specify warp_size when passing in the target: https://github.com/intel/intel-extension-for-pytorch/blob/dev/triton-test-3.0/intel_extension_for_pytorch/_inductor/xpu/triton_heuristics.py#L133. The warp_size GPUTarget field comes from upstream, can we autotune that instead?

I have upstreamed the threads_per_warp to the Triton in last year. https://github.com/triton-lang/triton/commit/45ba9af6ed1fdfea5a46a755be36c3319d82d249

I think #414 is also required to be upstreamed so that we can reuse the common lowering code.
I filed a another issue to track PR the #414 to upstream. https://github.com/intel/intel-xpu-backend-for-triton/issues/1135

I will double check the `warp_size` is used in AMD GPU backend for specific. We need to conform with Torch team whether they are agreed to use the warp size.

**Comment by whitneywhtsang** [comment]
> I think https://github.com/intel/intel-xpu-backend-for-triton/pull/414 is also required to be upstreamed so that we can reuse the common lowering code.

It may be difficult to upstream #414, as Triton upstream already have a way to represent threads per warp, which is per arch, instead of per kernel.

> We need to conform with Torch team whether they are agreed to use the warp size.

warp size of 32 is already passed by Pytorch/IPEX to Triton when creating `GPUTarget` for Intel.

**Comment by chengjunlu** [comment]
> And we should reevaluate our addition of `threads_per_warp` kernel operand, as now there are two ways of specifying the same thing. Currently we just ignore `warp_size`.

Let me split the PR. One PR to update the micro-bench code. And one PR to review the changes for autotune the `threads_per_warp`
